java/ql/src/experimental/Security/CWE/CWE-094/JexlInjection.qhelp

```diff
@@ -0,0 +1,65 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>
+Java EXpression Language (JEXL) is a simple expression language
+provided by the Apache Commons JEXL library.
+The syntax is close to a mix of ECMAScript and shell-script. 
+The language allows invocation of methods available in the JVM.
+If a JEXL expression is built using attacker-controlled data, 
+and then evaluated, then it may allow the attacker to run arbitrary code.
+</p>
+</overview>
+
+<recommendation>
+<p>
+Including untrusted input in a JEXL expression should be avoided. If it is not possible,
+JEXL expressions should be run in a sandbox that allows accessing only
+explicitly allowed classes.
+</p>
+</recommendation>
+
+<example>
+<p>
+The following example uses untrusted data to build and run a JEXL expression.
+</p>
+<sample src="UnsafeJexlExpressionEvaluation.java" />
+
+<p>
+The next example shows how an untrusted JEXL expression can be run
+in a sandbox that allows accessing only methods in the <code>java.lang.Math</code> class.
+The sandbox is implemented using <code>JexlSandbox</code> class that is provided by
+Apache Commons JEXL 3.
+However, it's recommended to avoid using untrusted input in JEXL expressions.
```

Comment:
This is a general recommendation that applies to any type of injection. Similar recommendations can be found in many docs, for example

https://github.com/github/codeql/blob/main/java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.qhelp#L24
https://github.com/github/codeql/blob/main/java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.qhelp#L24

If you don't find it useful, I'll just remove it. The `<recommendation>` section contains a similar recommendation. I'd prefer to keep it there.

Reply:
Note the one you linked to says what to do instead -- use a different format, because Java serialized blobs aren't suitable for untrusted data exchange. What are you saying here -- that JEXL is fundamentally unsuited to customisation with untrusted data, like the Java serialization format, or just that you have to sanitize your inputs, like constructing a SQL query?

Reply:
This is a common recommendation that suggest avoiding untrusted input in JEXL expressions. Not more. I think this is a valid recommendation even if it doesn't tell what to do instead :) Other paragraphs describe sandboxing as a measure to avoid injections.

Reply:
OK, sounds good -- perhaps simply change "it's" to "it is generally" to get across the idea that this is conventional wisdom not just us

Reply:
Ah, yesterday I removed this phrase in the `<example>` section. But there is a similar one in `<recommendation>`. I've [updated](https://github.com/github/codeql/pull/4965/commits/7cc7ec962e8f12e2869871fb878895894008bbd8) it with your suggestion.